internal/cmdupdate/cmdupdate.go

```diff
@@ -88,9 +90,12 @@ func (r *Runner) preRunE(_ *cobra.Command, args []string) error {
 		return err
 	}
 
-	relPath := string(p.DisplayPath)
 	r.Update.FullPackagePath = string(p.UniquePath)
 
+	relPath, err := resolveRelPath(p.UniquePath)
+	if err != nil {
+		return err
+	}
 	if strings.HasPrefix(relPath, pkg.ParentDir) {
 		return errors.Errorf("package path must be under current working directory")
```

Comment:
In english, what is it that's being tested here?

Also, what do you expect the user input PKG_PATH to be? Can it be absolute or just relative? Where's the documentation for the semantic of this arg? @phanimarupaka 

If PKG_PATH is absolute: In general, when there are symlinks present (let's say in one of the sub-packages), we no longer have a hierarchical fs. So simply doing a `HasPrefix` on the UniquePath won't be correct.

If PKG_PATH is relative: then this check is easier.

This is the reason I strongly suggest against doing any path-related computation outside of a centralized, well-encapsulated library. It's subtle and error prone.

Reply:
As mentioned before, not blocking this PR, but just assume this is buggy for now until we do it properly in future PRs.

Reply:
I think there are several questions here:

 * This functionality enforces that it is only possible to get/update package in subdirectories of cwd. We should decide if this check is needed.
   * If it is, then we should probably enforce it for all commands in a single place, so probably in the top-level kpt cobra command.
 * The provided path can be either absolute or relative, but we want the internals of kpt to work with absolute paths. We should handle this conversion as early as possible and have a single library for handling it.
 * Appropriate handling of symlinks is something we need to look at for all functionality in kpt. This probably includes understanding how it is handled in git.